src/pubsub.ts

```diff
@@ -182,26 +182,29 @@ export class SubscriptionManager {
             // convention this is the value returned from the mutation
             // resolver
             const onMessage = (rootValue) => {
-                let contextPromise;
-                if (typeof options.context === 'function') {
-                    contextPromise = new Promise((resolve) => {
-                        resolve(options.context());
-                    });
-                } else {
-                    contextPromise = Promise.resolve(options.context);
-                }
-                contextPromise.then((context) => {
-                    if (!filter(rootValue, context)) {
+                Promise.resolve().then(() => {
```

Comment:
I think you can optimize this code and do a better use of Promise chaining - this way you can avoid the inner promises and the empty `resolve()`. you can also merge the two `catch` that does the same thing.